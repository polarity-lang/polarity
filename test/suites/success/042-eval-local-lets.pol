data Nat {
    Z,
    S(x: Nat)
}

data Eq(a: Type, x y: a) {
    Refl(a: Type, x: a): Eq(a, x, x),
}

def Nat.add(y: Nat): Nat {
    Z => y,
    S(x') => S(x'.add(y))
}

#[transparent]
let two_plus_three: Nat {
    let x: Nat := 2;
    let y: Nat := 3;
    x.add(y)
}

let two_plus_three_equals_five: Eq(Nat, two_plus_three, 5) {
    Refl(Nat, 5)
}