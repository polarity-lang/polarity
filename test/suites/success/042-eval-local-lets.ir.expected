def .add(y) {
    Z => y,
    S(x') => S(x'.add(y)),
}

let two_plus_three {
    let x := S(S(Z));
    let y := S(S(S(Z)));
    x.add(y)
}

let two_plus_three_equals_five { Refl(S(S(S(S(S(Z)))))) }
